### PR TITLE
Analyzer

### PR DESCRIPTION
**Pull request by pbortnik**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/events/handler/ProjectActivityHandler.java [hunk 1]
```diff
@@ -87,10 +93,36 @@ public void onApplicationEvent(ProjectUpdatedEvent event) {
...
+			processElasticParameters(history, NUMBER_OF_LOG_LINES, project.getConfiguration().getAnalyzerConfig().getNumberOfLogLines(),
+					analyzerConfig.getNumberOfLogLines()
+			);
+		}
+	}
+
+	private void processElasticParameters(List<Activity.FieldValues> history, String elasticParameterName, Integer oldValue,
+			Integer newValue) {
+		ofNullable(newValue).ifPresent(param -> {
+			if (!newValue.equals(oldValue)) {
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/events/handler/ProjectActivityHandler.java]: newValue should be replaced with param

File: src/main/java/com/epam/ta/reportportal/core/analyzer/impl/AnalyzerStatusCache.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+	private LoadingCache<String, String> analyzerStatus;
+
+	public AnalyzerStatusCache() {
+		analyzerStatus = CacheBuilder.newBuilder()
+				.maximumSize(MAXIMUM_SIZE)
+				.expireAfterWrite(CACHE_ITEM_LIVE, TimeUnit.MINUTES)
+				.build(new CacheLoader<String, String>() {
+					@Override
+					public String load(String key) throws Exception {
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/analyzer/impl/AnalyzerStatusCache.java]: to remove

File: src/main/java/com/epam/ta/reportportal/core/project/impl/UpdateProjectHandler.java [hunk 3]
```diff
@@ -175,13 +188,18 @@ private void processConfiguration(ProjectConfiguration modelConfig, Project.Conf
...
 
-		if (null != modelConfig.getIsAutoAnalyzerEnabled()) {
-			dbConfig.setIsAutoAnalyzerEnabled(modelConfig.getIsAutoAnalyzerEnabled());
-		}
-
-		if (null != modelConfig.getAnalyzerMode()) {
-			dbConfig.setAnalyzerMode(AnalyzeMode.fromString(modelConfig.getAnalyzerMode()));
-		}
+		ofNullable(modelConfig.getAnalyzerConfig()).ifPresent(analyzerConfig -> {
+			expect(analyzerStatusCache.getAnalyzerStatus().asMap().containsKey(projectName), equalTo(false)).verify(
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/project/impl/UpdateProjectHandler.java]: bug

File: src/main/java/com/epam/ta/reportportal/ws/converter/converters/LaunchConverter.java [hunk 4]
```diff
@@ -50,10 +53,14 @@ private LaunchConverter() {
 		resource.setTags(db.getTags());
 		resource.setMode(db.getMode());
 		resource.setApproximateDuration(db.getApproximateDuration());
-		resource.setIsProcessing(false);
+		resource.setIsProcessing(analyzerStatusCache.getAnalyzerStatus().asMap().containsKey(resource.getLaunchId()));
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/ws/converter/converters/LaunchConverter.java]: remove as map

File: src/main/java/com/epam/ta/reportportal/events/handler/ProjectActivityHandler.java (hunk 1 above)
Comment (pbortnik) [reply src/main/java/com/epam/ta/reportportal/events/handler/ProjectActivityHandler.java]: +

File: src/main/java/com/epam/ta/reportportal/core/analyzer/impl/AnalyzerStatusCache.java (hunk 2 above)
Comment (pbortnik) [reply src/main/java/com/epam/ta/reportportal/core/analyzer/impl/AnalyzerStatusCache.java]: +

File: src/main/java/com/epam/ta/reportportal/core/project/impl/UpdateProjectHandler.java (hunk 3 above)
Comment (pbortnik) [reply src/main/java/com/epam/ta/reportportal/core/project/impl/UpdateProjectHandler.java]: +

File: src/main/java/com/epam/ta/reportportal/ws/converter/converters/LaunchConverter.java (hunk 4 above)
Comment (pbortnik) [reply src/main/java/com/epam/ta/reportportal/ws/converter/converters/LaunchConverter.java]: +

**Pull request merged**